Add documentation for Analyze Experiment MCP/CLI command

nsthorat opened the pull request:
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/nsthorat/mlflow/pull/17978?quickstart=1)

#### Install mlflow from this PR

```
# mlflow
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/17978/merge
# mlflow-skinny
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/17978/merge#subdirectory=libs/skinny
```

For Databricks, use the following command:

```
%sh curl -LsSf https://raw.githubusercontent.com/mlflow/mlflow/HEAD/dev/install-skinny.sh | sh -s pull/17978/merge
```

</p>
</details>

- Add new Insights section under genai/eval-monitor
- Document both MCP (/analyze-experiment) and CLI (mlflow ai-commands genai/analyze-experiment) usage
- Include comprehensive workflow documentation covering setup, experiment selection, analysis, and report generation
- Update sidebarsGenAI.ts to include the new Insights section in navigation
- Cross-link to existing MCP setup documentation for prerequisites

### Related Issues/PRs

<!-- Uncomment 'Resolve' if this PR can close the linked items. -->
<!-- Resolve --> #xxx

### What changes are proposed in this pull request?

<!-- Please fill in changes proposed in this PR. -->

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

<!-- Attach code, screenshot, video used for manual testing here. -->

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [x] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [x] Instructions

### Release Notes

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

<!-- Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change. -->

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/prompts`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages

<!--
Insert an empty named anchor here to allow jumping to this section with a fragment URL
(e.g. https://github.com/mlflow/mlflow/pull/123#user-content-release-note-category).
Note that GitHub prefixes anchor names in markdown with "user-content-".
-->

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [x] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

nsthorat committed:
Add documentation for Analyze Experiment MCP/CLI command

- Add new Insights section under genai/eval-monitor
- Document both MCP (/analyze-experiment) and CLI (mlflow ai-commands genai/analyze-experiment) usage
- Include comprehensive workflow documentation covering setup, experiment selection, analysis, and report generation
- Update sidebarsGenAI.ts to include the new Insights section in navigation
- Cross-link to existing MCP setup documentation for prerequisites

Signed-off-by: nsthorat <nsthorat@gmail.com>

BenWilson2 commented on docs/docs/genai/eval-monitor/insights/analyze-experiment.mdx:
```diff
@@ -0,0 +1,127 @@
+
+Automatically analyze traces in your MLflow experiments to identify operational issues, quality problems, and performance patterns. The Analyze Experiment tool uses hypothesis-driven analysis to systematically examine your GenAI application's behavior and generate comprehensive markdown reports.
+
+## Overview
```
Might want to look to convert this section to using the FeatureHighlights component (title + description cards with simple icons) to conform to the "Why is this important for me to look at" section of many of the GenAI docs

BenWilson2 commented on docs/docs/genai/eval-monitor/insights/analyze-experiment.mdx:
```diff
@@ -0,0 +1,127 @@
+
+The Analyze Experiment functionality is available through two methods:
+
+<Tabs>
```
Let's use the `TabsWrapper` component around Tabs

BenWilson2 replied:
And we can also use the `TabItem` sections instead of putting md headers in the tab contents

BenWilson2 commented on docs/docs/genai/eval-monitor/insights/analyze-experiment.mdx:
````diff
@@ -0,0 +1,127 @@
+
+If you have MLflow's MCP server configured, you can simply run:
+
+```
````
don't forget language indicator for formatting `bash`

nsthorat replied:
done

BenWilson2 commented on docs/docs/genai/eval-monitor/insights/analyze-experiment.mdx:
```diff
@@ -0,0 +1,127 @@
+</Tabs>
+
+## Analysis Workflow
+
```
It would be nice to show a visual component to the devloop here. Check out the `DAGLoop` component :)

BenWilson2 commented on docs/docs/genai/eval-monitor/insights/analyze-experiment.mdx:
```diff
@@ -0,0 +1,127 @@
+
+The tool will ask you to configure authentication:
+
+- **Databricks**: Provide workspace URL and personal access token, or use a Databricks CLI profile
```
Each of the sections here might look a little repeated. Can some of the elements be broken up and rendered as something other than bulleted lists? Take a look at the tracing docs or the scorers docs to see examples to get inspiration from

dbczumar commented on docs/docs/genai/eval-monitor/insights/analyze-experiment.mdx:
```diff
@@ -0,0 +1,127 @@
+
+# Analyze Experiment
+
+Automatically analyze traces in your MLflow experiments to identify operational issues, quality problems, and performance patterns. The Analyze Experiment tool uses hypothesis-driven analysis to systematically examine your GenAI application's behavior and generate comprehensive markdown reports.
```
```suggestion
Automatically analyze traces in your MLflow experiments to find operational issues, quality problems, and performance patterns. The Analyze Experiment tool uses hypothesis-driven analysis to systematically examine your GenAI application's behavior, identify the most important problems, and create a plan for addressing them in the form of a comprehensive markdown report.
```

dbczumar commented on docs/docs/genai/eval-monitor/insights/analyze-experiment.mdx:
```diff
@@ -0,0 +1,127 @@
+import TabItem from "@theme/TabItem"
+import { MessageSquare, Target, BarChart3, ExternalLink, Bot, Users, CheckCircle, TrendingUp, Database, Book, Rocket, Scale } from "lucide-react";
+
+# Analyze Experiment
```
@nsthorat Wondering if there's potentially a more eye-catching / attention-grabbing name to use in the docs sidebar. Something that makes it 100% obvious that this feature automatically identifies the most important issues and creates a prioritized plan for which ones to address next

nsthorat committed:
Address all PR review comments for AI Issue Discovery documentation

- Fix CLI command to use underscore: genai/analyze_experiment
- Add note that CLI must run inside coding agent with MCP support
- Update description per dbczumar's suggestion to emphasize problem identification and planning
- Convert Overview section to use FeatureHighlights component with icon cards
- Add TabsWrapper component around Tabs for consistency
- Add DAGLoop visual workflow diagram with 5 interactive steps
- Convert Report Content section to use ConceptOverview component
- Rename page title from 'Analyze Experiment' to 'AI Issue Discovery'
- Rename sidebar category from 'Insights' to 'AI Insights'
- Rename file and folder structure to match new naming:
  - insights/analyze-experiment.mdx -> ai-insights/ai-issue-discovery.mdx

Addresses all 9 review comments from PR #17978

Signed-off-by: nsthorat <nsthorat@gmail.com>

dbczumar commented on docs/docs/genai/eval-monitor/ai-insights/ai-issue-discovery.mdx:
```diff
@@ -0,0 +1,189 @@
+
+- **Databricks**: Provide workspace URL and personal access token, or use a Databricks CLI profile
+- **Local MLflow**: Specify tracking URI (SQLite, PostgreSQL, MySQL, or file store)
+- **Environment Variables**: Use existing MLflow environment configuration
```
Nit - can we make this a bit clearer? Not 100% obvious what `existing MLflow environment configuration` is

nsthorat replied:
done! linked to the env docs

dbczumar reviewed (approved):
LGTM with one small nit. @nsthorat can we also add a screenshot or two?

nsthorat committed:
Add screenshots and clarify environment variables documentation

- Add ImageBox component and 5 workflow screenshots to AI Issue Discovery page
- Clarify MLflow environment variables with specific example and link to setup guide
- Screenshots illustrate each step: authentication, experiment selection, agent purpose, hypothesis analysis, and report generation
- Addresses PR review feedback about visual improvements and environment variable clarity

Co-authored-by: dbczumar <dbczumar@databricks.com>
Signed-off-by: nsthorat <nsthorat@gmail.com>

nsthorat committed:
Update AI Issue Discovery screenshots with actual tool screenshots

- Replace placeholder images with actual screenshots from running the tool
- Step 1: Setup and authentication dialog
- Step 2: Experiment selection interface
- Step 3: Agent purpose identification
- Step 4: Hypothesis-driven analysis in progress
- Step 5: Generated report with insights and recommendations

Signed-off-by: nsthorat <nsthorat@gmail.com>

nsthorat commented:
Added a screenshot for each step :)

dbczumar committed:
Fix prettier formatting for ai-issue-discovery.mdx

Signed-off-by: dbczumar <corey.zumar@databricks.com>
